src/pg_timestamp.erl

```diff
@@ -49,9 +49,24 @@ encode_timestamp(infinity) ->
     16#7FFFFFFFFFFFFFFF;
 encode_timestamp('-infinity') ->
     -16#8000000000000000;
+encode_timestamp({{Year, Month, Day}, {Hour, Minute, Seconds}, {HourOffset, MinuteOffset}}) when is_integer(Seconds) ->
```

Comment:
Instead of just checking if its an integer we can check if its between -12 and 14 so the user gets a quick error response from the library that they have an incorrect timestamp.

Reply:
Should also verify `MinuteOffset` is `<` 60. I guess we can do the same for all the other values except for Year but you don't have to worry about that for this PR.

Reply:
But then why is the type you specify restricted?

Reply:
I guess we don't have to? I don't really understand what that type spec is for.

In the non-tz case, there is no restriction on hour, minute, second so I could just make them `integer()` for the offsets too? 

I guess I am unsure what they should represent, should they be timestamps you would actually get from PG or just valid inputs to the functions in this package? As far as I can tell, they are unused locally?

Reply:
They are for valid input. It is to give the user information when the input is invalid, it'll print the valid type spec for the type they are trying to encode.

Reply:
Updated in [ee0a639](https://github.com/tsloughter/pg_types/pull/10/commits/ee0a639f4bd7c8a3adeed33015c33acf4f4c548b)

Reply:
548904a44c908237bd99cb88af54b783107c836d fixed. If hour offset was zero and minute offset was negative this code would fail. It is probably simplest to enforce the minute offset to be positive.